dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultAnalyticsService.java

```diff
@@ -183,10 +190,30 @@
     @Autowired
     private DataQueryService dataQueryService;
 
+    @Autowired
+    private DhisConfigurationProvider dhisConfig;
+
+    @Autowired
+    private CacheProvider cacheProvider;
+
     // -------------------------------------------------------------------------
     // AnalyticsService implementation
     // -------------------------------------------------------------------------
 
+    private Cache<Grid> queryCache;
+
+    @PostConstruct
+    public void init()
+    {
+        Long expiration = Long.parseLong( dhisConfig.getProperty( ConfigurationKey.ANALYTICS_CACHE_EXPIRATION ) );
+        boolean enabled = expiration > 0 && !SystemUtils.isTestRun();
+
+        queryCache = cacheProvider.newCacheBuilder( Grid.class ).forRegion( "analyticsQueryResponse" )
```

Comment:
I need to double check on what happens when the Value type is specified as an interface (Grid). I remember doing some tests back when I implemented cacheProvider and I recall  that it worked fine, because it stores the concrete class information along with the value json.

Reply:
Cool.